### ISSUE TITLE
NREL5MW: Failure (apparently) at start of time step 2 due to memory allocation error

### ISSUE THREAD

**Issue by jhux2** [Exawind/nalu-wind#926]
While running with the unrefined NREL5MW mesh on 18 MPI ranks of Summit (3 nodes), my job fails at the start of Time Step 2 with the error

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Kokkos failed to allocate memory for label "Kokkos::CudaSpace::TeamScratchMemory".  Allocation using MemorySpace named "Cuda" failed with the following error:  Allocation of size 1.279 G failed, likely due to insufficient memory.  (The allocation mechanism was cudaMalloc().  The Cuda allocation returned the error code ""cudaErrorMemoryAllocation".)
```

Has anyone seen a similar failure?

I'm using Trilinos VOTD `9ebcf3e` and Nalu-Wind `8e8c8f6`.

 Here's the runlog and gdb backtrace.

<details>

<summary>
run log
</summary>

```
        ContinuityEQS         24          0.261728        5.63033       2.70161
 1/1ShearStressTransportWrap

*******************************************************
***** Belos Iterative Solver:  Pseudo Block Gmres
***** Maximum Iterations: 200
***** Block Size: 1
***** Residual Test:
*****   Test 1 : Belos::StatusTestImpResNorm<>: (2-Norm Res Vec) / (2-Norm Prec Res0), tol = 1e-05
*******************************************************
Iter   0, [ 1] :    1.000000e+00
Iter   1, [ 1] :    2.614061e-04
Iter   2, [ 1] :    1.259681e-05
Iter   3, [ 1] :    1.881636e-06
        TurbKineticEnergyEQS   3         0.0604296        6.69333   0.000428681

*******************************************************
***** Belos Iterative Solver:  Pseudo Block Gmres
***** Maximum Iterations: 200
***** Block Size: 1
***** Residual Test:
*****   Test 1 : Belos::StatusTestImpResNorm<>: (2-Norm Res Vec) / (2-Norm Prec Res0), tol = 1e-05
*******************************************************
Iter   0, [ 1] :    1.000000e+00
Iter   1, [ 1] :    5.048229e-08
        SpecDissRateEQS        1        4.9803e+06     2.0561e+10       60.0722
Mass Balance Review:
Density accumulation: 0
Integrated inflow:    -15312500
Integrated open:      15312494.36104351
Total mass closure:   -5.638956487178802
Filter Size 0.009101499927188001
Mean System Norm: 4112201288.336813 1 0.00455075
WallClockTime: 1 Pre: 35.5632 NLI: 185.141 Post: 0.237246 Total: 220.942
*******************************************************
Time Step Count: 2 Current Time: 0.0091015
 dtN: 0.00455075 dtNm1: 0.00455075 gammas: 1.5 -2 0.5
 DualNodalVolume min: 3.71294e-10 max: 39763.6 total: 2.92973e+09
 Max Courant: 7623.65 Max Reynolds: 7.39241e+06 (realm_1)
   Realm Nonlinear Iteration: 1/1

realm_1::advance_time_step()
NLI    Name           Linear Iter      Linear Res     NLinear Res    Scaled NLR
---    ----           -----------      ----------     -----------    ----------
1/4    Equation System Iteration
 1/1      myLowMach
```

</details>

<details>

<summary>
backtrace
</summary>

```
#0  0x0000200040383618 in raise () from /lib64/power9/libc.so.6
#1  0x0000200040363a2c in abort () from /lib64/power9/libc.so.6
#2  0x0000200040054a98 in __gnu_cxx::__verbose_terminate_handler() () from /lib64/libstdc++.so.6
#3  0x000020004004f6d4 in __cxxabiv1::__terminate(void (*)()) () from /lib64/libstdc++.so.6
#4  0x000020004004f790 in std::terminate() () from /lib64/libstdc++.so.6
#5  0x000020004004fc6c in __cxa_throw () from /lib64/libstdc++.so.6
#6  0x000000001688fb2c in Kokkos::Impl::throw_runtime_exception (msg=...) at /usr/include/c++/8/ext/new_allocator.h:86
#7  0x0000000016894348 in Kokkos::Impl::safe_throw_allocation_with_header_failure (space_name="Cuda", label="Kokkos::CudaSpace::TeamScratchMemory", failure=...) at /usr/include/c++/8/bits/basic_string.h:1927
#8  0x00000000168a35b0 in Kokkos::Impl::checked_allocation_with_header<Kokkos::CudaSpace> (space=..., label="Kokkos::CudaSpace::TeamScratchMemory", alloc_size=<optimized out>) at /usr/include/c++/8/ext/new_allocator.h:79
#9  0x00000000168a025c in Kokkos::Impl::SharedAllocationRecord<Kokkos::CudaSpace, void>::SharedAllocationRecord (this=0x15556f940, arg_space=..., arg_label="Kokkos::CudaSpace::TeamScratchMemory", arg_alloc_size=<optimized out>, arg_dealloc=<optimized out>)
    at /ccs/home/jjhu/exawind/sources/Trilinos-dev/packages/kokkos/core/src/Cuda/Kokkos_CudaSpace.cpp:528
#10 0x00000000168a3688 in Kokkos::Impl::SharedAllocationRecordCommon<Kokkos::CudaSpace>::allocate (arg_space=..., arg_label="Kokkos::CudaSpace::TeamScratchMemory", arg_alloc_size=1373634560)
    at /ccs/home/jjhu/exawind/sources/Trilinos-dev/packages/kokkos/core/src/impl/Kokkos_SharedAlloc_timpl.hpp:63
#11 0x00000000168a37b0 in Kokkos::Impl::SharedAllocationRecordCommon<Kokkos::CudaSpace>::reallocate_tracked (arg_alloc_ptr=<optimized out>, arg_alloc_size=1373634560)
    at /ccs/home/jjhu/exawind/sources/Trilinos-dev/packages/kokkos/core/src/impl/Kokkos_SharedAlloc_timpl.hpp:99
#12 0x00000000168a55b8 in Kokkos::kokkos_realloc<Kokkos::CudaSpace> (arg_alloc_size=<optimized out>, arg_alloc=<optimized out>) at /ccs/home/jjhu/exawind/sources/Trilinos-dev/packages/kokkos/core/src/Kokkos_Core.hpp:229
#13 Kokkos::Impl::CudaInternal::resize_team_scratch_space (this=<optimized out>, bytes=1373634560, force_shrink=<optimized out>) at /ccs/home/jjhu/exawind/sources/Trilinos-dev/packages/kokkos/core/src/Cuda/Kokkos_Cuda_Instance.cpp:712
#14 0x000000001030d718 in Kokkos::Impl::ParallelFor<__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>), &(void sierra::nalu::AssembleFaceElemSolverAlgorithm::run_face_elem_algorithm<__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> >(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>)), 1u>, void (Kokkos::Impl::CudaTeamMember const&), stk::NgpVector<unsigned int> const, stk::mesh::DeviceMesh const, stk::topology::rank_t, int, sierra::nalu::ElemDataRequestsGPU, sierra::nalu::ElemDataRequestsGPU, unsigned int const, unsigned int const, int, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> >, Kokkos::TeamPolicy<Kokkos::Cuda>, Kokkos::Cuda>::ParallelFor(__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>), &(void sierra::nalu::AssembleFaceElemSolverAlgorithm::run_face_elem_algorithm<__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> >(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>)), 1u>, void (Kokkos::Impl::CudaTeamMember const&), stk::NgpVector<unsigned int> const, stk::mesh::DeviceMesh const, stk::topology::rank_t, int, sierra::nalu::ElemDataRequestsGPU, sierra::nalu::ElemDataRequestsGPU, unsigned int const, unsigned int const, int, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> > const&, Kokkos::TeamPolicy<Kokkos::Cuda> const&) (arg_policy=..., arg_functor=..., this=0x7fffc242c6b0)
    at /gpfs/alpine/scratch/jjhu/cfd116/exawind/install/trilinos-gcc-8.3.1-cuda-11.0.3-spectrum-mpi-10.4.0.3-relwithdebinfo-a5a5bf9a/include/Kokkos_Cuda.hpp:249
#15 Kokkos::parallel_for<Kokkos::TeamPolicy<Kokkos::Cuda>, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>), &(void sierra::nalu::AssembleFaceElemSolverAlgorithm::run_face_elem_algorithm<__nv_hdl_wrapper_t<false,--Type <RET> for more, q to quit, c to continue without paging--
 false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> >(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>)), 1u>, void (Kokkos::Impl::CudaTeamMember const&), stk::NgpVector<unsigned int> const, stk::mesh::DeviceMesh const, stk::topology::rank_t, int, sierra::nalu::ElemDataRequestsGPU, sierra::nalu::ElemDataRequestsGPU, unsigned int const, unsigned int const, int, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> > >(Kokkos::TeamPolicy<Kokkos::Cuda> const&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>), &(void sierra::nalu::AssembleFaceElemSolverAlgorithm::run_face_elem_algorithm<__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> >(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>)), 1u>, void (Kokkos::Impl::CudaTeamMember const&), stk::NgpVector<unsigned int> const, stk::mesh::DeviceMesh const, stk::topology::rank_t, int, sierra::nalu::ElemDataRequestsGPU, sierra::nalu::ElemDataRequestsGPU, unsigned int const, unsigned int const, int, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::enable_if<Kokkos::is_execution_policy<Kokkos::TeamPolicy<Kokkos::Cuda> >::value, void>::type*) (str="", functor=..., policy=...) at /gpfs/alpine/scratch/jjhu/cfd116/exawind/install/trilinos-gcc-8.3.1-cuda-11.0.3-spectrum-mpi-10.4.0.3-relwithdebinfo-a5a5bf9a/include/Kokkos_Parallel.hpp:139
#16 sierra::nalu::AssembleFaceElemSolverAlgorithm::run_face_elem_algorithm<__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff> >(stk::mesh::BulkData&, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (sierra::nalu::AssembleFaceElemSolverAlgorithm::*)(), &sierra::nalu::AssembleFaceElemSolverAlgorithm::execute, 1u>, void (sierra::nalu::SharedMemData_FaceElem<Kokkos::Impl::CudaTeamMember, Kokkos::ScratchMemorySpace<Kokkos::Cuda> >&), unsigned long const, Kokkos::View<sierra::nalu::nalu_ngp::NGPCopyHolder<sierra::nalu::Kernel>*, Kokkos::LayoutRight, Kokkos::CudaSpace>, unsigned int const, unsigned int const, double, sierra::nalu::NGPApplyCoeff>) (this=<optimized out>, bulk=..., lamdbaFunc=...) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/include/AssembleFaceElemSolverAlgorithm.h:80
#17 0x0000000010309f00 in sierra::nalu::AssembleFaceElemSolverAlgorithm::execute (this=0x255c75d0)
    at /gpfs/alpine/scratch/jjhu/cfd116/exawind/install/trilinos-gcc-8.3.1-cuda-11.0.3-spectrum-mpi-10.4.0.3-relwithdebinfo-a5a5bf9a/include/impl/Kokkos_SharedAlloc.hpp:494
#18 0x0000000010598608 in sierra::nalu::SolverAlgorithmDriver::execute (this=0x35f91e10) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/SolverAlgorithmDriver.C:120
#19 0x00000000103cd26c in sierra::nalu::EquationSystem::assemble_and_solve (this=0x47903160, deltaSolution=0x4408c730) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/EquationSystem.C:296
#20 0x000000001047088c in sierra::nalu::MomentumEquationSystem::assemble_and_solve (this=0x47903160, deltaSolution=0x4408c730) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/LowMachEquationSystem.C:2599
#21 0x000000001046fce8 in sierra::nalu::LowMachEquationSystem::solve_and_update (this=0x4d0be7b0) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/LowMachEquationSystem.C:722
#22 0x00000000103d8b48 in sierra::nalu::EquationSystems::solve_and_update (this=0x4c2e7710) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/EquationSystems.C:743
#23 0x0000000010550968 in sierra::nalu::Realm::advance_time_step (this=0x4c2e74b0) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/Realm.C:1787
#24 0x00000000105b7dec in sierra::nalu::TimeIntegrator::integrate_realm (this=0x44138020) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/TimeIntegrator.C:342
#25 0x0000000010580f34 in sierra::nalu::Simulation::run (this=0x7fffc2445948) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/src/Simulation.C:173
#26 0x00000000100f7338 in main (argc=<optimized out>, argv=<optimized out>) at /ccs/home/jjhu/exawind/sources/nalu-wind-exawind/nalu.C:178
```

</details>

**Comment by alanw0** [comment]
This sounds like the same error that's been killing several of our nightly tests. It started end of October with a trilinos update that included a kokkos update. We finally tracked it down to the specific kokkos commit and filed it as a kokkos github issue. The kokkos folks provided a small patch which fixes it. Jon has added that patch to spack-manager. If you want to apply it yourself, it is in the comments for kokkos github issue 4665.

**Comment by jhux2** [comment]
@alanw0 Thanks, I'll give that a try!

**Comment by jhux2** [comment]
@alanw0 The patch suggested in https://github.com/kokkos/kokkos/issues/4665 works, thank you.

**Comment by jhux2** [comment]
When the Kokkos patch merges into Trilinos develop, I'll close this issue.

**Comment by jhux2** [comment]
Fixed in Kokkos 3.6.0 promotion (https://github.com/trilinos/Trilinos/pull/10253).
